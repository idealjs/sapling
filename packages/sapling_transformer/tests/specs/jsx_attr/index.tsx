function Attr() {
  return <button type="button">attr</button>;
}

let someStyle = { color: "red", fontSize: "16px" };

function Attr2() {
  return <div style={someStyle}>attr</div>;
}
