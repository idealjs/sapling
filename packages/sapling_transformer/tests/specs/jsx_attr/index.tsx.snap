---
source: packages/sapling_transformer/tests/test_utils.rs
expression: index.tsx
---
# Input
```tsx
function Attr() {
  return <button type="button">attr</button>;
}

let someStyle = { color: "red", fontSize: "16px" };

function Attr2() {
  return <div style={someStyle}>attr</div>;
}

```

# Transformations
```tsx
function Attr() {
  return _$createJsxTagElement(() => {
    let _el$ = _$createElement("button");
    _$setProp(_el$, "type", "button");
    _$insertNode(_el$, _$createTextNode("attr"));
    return _el$;
  });
}

let someStyle = { color: "red", fontSize: "16px" };

function Attr2() {
  return _$createJsxTagElement(() => {
    let _el$ = _$createElement("div");
    _$setProp(_el$, "style", someStyle);
    _$insertNode(_el$, _$createTextNode("attr"));
    return _el$;
  });
}

```
